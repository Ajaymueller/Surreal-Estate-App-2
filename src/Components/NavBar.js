import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import FacebookLogin from "react-facebook-login";
import "../styles/NavBar.css";

const LogoutButton = ({ onLogout }) => (
  <button type="button" onClick={onLogout} data-testid="sign-out">
    Logout
  </button>
);

const NavBar = ({ onLogin, onLogout, userID }) => (
  <div className="navbar" data-testid="navbar">
    <Link to="/"><img
      className="navbar-logo"
      src="https://mcrcodes.s3.eu-west-2.amazonaws.com/course/surreal-estate/logo.png"
      alt="logo"
    /> </Link>
    <ul className="navbar-links">
      <li className="navbar-links-item">
        <Link to="/properties" data-testid="view-properties-id">
          View Properties</Link>
      </li>
      <li className="navbar-links-item">
        <Link to="/saved-properties" data-testid="saved-properties">
          Saved Properties</Link>
      </li>
      <li className="navbar-links-item">
        <Link to="/add-property" data-testid="add-property-id">
          Add Property</Link>
      </li>
      {!userID ? (
        <FacebookLogin
          appId="2346104855684184"
          autoLoad
          fields="name,email,picture"
          // onClick={componentClicked}
          callback={onLogin}
        />
      ) : (
        <LogoutButton onLogout={onLogout} />
      )}
    </ul>
  </div>
);

LogoutButton.propTypes = {
  onLogout: PropTypes.func.isRequired,
};

NavBar.propTypes = {
  onLogin: PropTypes.func.isRequired,
  onLogout: PropTypes.func.isRequired,
  userID: PropTypes.string.isRequired,
};

export default NavBar;