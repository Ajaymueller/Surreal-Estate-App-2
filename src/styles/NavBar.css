
.navbar {
    background-color: #ffffff;
    border-bottom: 1px solid grey;
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-start;
    align-items: center;
}
  
  .navbar-logo {
    max-width: 200px;
    flex: 0 1 auto;
  }
  
  .navbar-links {
    margin: 0;
    padding: 0;
    display: flex;
    flex-direction: row;
    justify-content: end;
  }
  
  .navbar-links-item {
    margin: 0;
    padding: 24px;
    display: inline-block;
  }
  
  .navbar-links-item > * {
    text-decoration: none;
  }