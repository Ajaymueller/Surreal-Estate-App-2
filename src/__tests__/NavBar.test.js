import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import NavBar from '../Components/NavBar';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';

const props = {
    onLogin: jest.fn(), 
    userID: 10, 
    onLogout: jest.fn(),
};

describe("NavBar", () => {
    it("component renders with correct props", () => {
    const { asFragment } = render( 
    <MemoryRouter><NavBar {...props} /></MemoryRouter>)
    expect(asFragment).toMatchSnapshot();
    });
    it("displays a logo", () => {
    const { getByAltText } = render( 
    <MemoryRouter><NavBar {...props} /></MemoryRouter>)
    const logo = getByAltText("logo");
    expect(logo).toBeInTheDocument();
    })
    it("displays a list with link to properties, add properties page and saved properties page", () => {
    const { getByText } = render( 
    <MemoryRouter><NavBar {...props} /></MemoryRouter>)
    const viewProperties = getByText("View Properties");
    const addProperty = getByText("Add Property");
    const savedProperties = getByText("Saved Properties");
    expect(viewProperties).toBeInTheDocument();
    expect(addProperty).toBeInTheDocument();
    expect(savedProperties).toBeInTheDocument();
    });
    it("displays a sign out button if there is a userID", () => {
    const { getByTestId } = render( 
    <MemoryRouter><NavBar {...props} /></MemoryRouter>)
    const button = getByTestId("sign-out");
    expect(button).toBeInTheDocument();
    });
});

describe("with router", () => {
    xit("should navigate to view properties page with /properties", () => {
        const { getByTestId, getByText } = 
        render( 
        <MemoryRouter><NavBar {...props} /></MemoryRouter>)
        const viewProperties = getByTestId("view-properties-id")
        fireEvent.click(getByText("View Properties"))
        expect(viewProperties).toBeInTheDocument();
        expect(viewProperties).toHaveAttribute("href", "/properties");
    });
    xit("should navigate to Add properties page with /add-property", () => {
        const { getByTestId, getByText } = 
        render( 
        <MemoryRouter><NavBar {...props} /></MemoryRouter>)
        const addProperty = getByTestId("add-property-id")
        fireEvent.click(getByText("Add Property"))
        expect(addProperty).toBeInTheDocument();
        expect(addProperty).toHaveAttribute("href", "/add-property");
    });
})