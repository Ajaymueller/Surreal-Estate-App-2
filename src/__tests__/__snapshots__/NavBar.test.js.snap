exports[`NavBar component renders with correct props 1`] = `[Function]`;
